### PR TITLE
[Docs] Add terms to glossary

### PR DESCRIPTION
**Pull request by swarnap**
Expanding glossary to include industry terms to help with SEO ranking.

### CODE REVIEW

File: website/pages/docs/install/glossary.mdx [hunk 1]
```diff
@@ -74,3 +74,333 @@ over the internet or wide area network.
...
+
+## Reverse Proxy	
+A reverse proxy handles requests coming from outside, to the internal 
+network. Reverse Proxy provides a level of security that prevents the 
+external clients from having direct access to data on the corporate servers. 
+The reverse proxy is usually placed between the web server and the external 
+traffic.  
+See also [Forward Proxy](link to that glossary term).
+
+## Role-based Access Controls	"The act of restricting or provisioning access
```
Comment (jsosulska) [inline website/pages/docs/install/glossary.mdx:335]: Seems  like a missed formatting here.

Comment (swarnap) [reply website/pages/docs/install/glossary.mdx:335]: I meant to indicate that we should cross-link the terms to that specific spot on the page.  So when folks are looking at the definition of Reverse Proxy, we provide them a link to the definition of Forward Proxy on the same page.  Does that help?

File: website/pages/docs/install/glossary.mdx [hunk 2]
```diff
@@ -74,3 +74,333 @@ over the internet or wide area network.
 
 Remote Procedure Call. This is a request / response mechanism allowing a
 client to make a request of a server.
+
+# Glossary
```
Comment (preetapan) [inline website/pages/docs/install/glossary.mdx:78]: Do we need this header? I am looking at the preview in https://deploy-preview-9348--consul-docs-preview.netlify.app/docs/install/glossary and this page already has a header "Consul Glossary"

File: website/pages/docs/install/glossary.mdx [hunk 3]
```diff
@@ -74,3 +74,333 @@ over the internet or wide area network.
...
+## Application Services	
+Application Services are a group of services, such as application performance 
+monitoring, load balancing, service discovery, service proxy, security, 
+autoscaling, etc. needed to deploy, run, and improve applications. 
+
+## Authentication and Authorization (AuthN and AuthZ)	
+Authentication (AuthN) deals with establishing user identity while Authorization
+(AuthZ) allows or denies access to the user based on user identity.
+
+See also [Identity-based Authorization](link to that glossary term).
```
Comment (swarnap) [inline website/pages/docs/install/glossary.mdx:111]: Removing these.

File: website/pages/docs/install/glossary.mdx (hunk 2 above)
Comment (swarnap) [reply website/pages/docs/install/glossary.mdx]: I updated this to "Consul Vocabulary" as these are the terms that are used specifically in the context of the product.  The rest of the terms I included in the list are used in the broader context of the conversation around networking and microservices.

**Review by swarnap — COMMENTED** [review]
I updated the section headers (Vocabulary and Glossary) along with the blurb of descriptions for each of these sections.  I also addressed the comment around inter-linking to Glossary terms on the same page. 

It looks like this is ready for a review, preview, and then deploy (if layout looks good).

**Comment by jsosulska** [comment]
Hi @swarnap  - Looks like you have a merge conflict that needs to be resolved. I also added Consul Docs wrt the location and terms.

**Comment by jsosulska** [comment]
Close in favor of rebase in #10074

**Pull request closed without merging**